docs/src/literate-tutorials/dg_heat_equation.jl

```diff
@@ -0,0 +1,334 @@
+#     ```math
+#      \int_\Omega \nabla u \cdot \nabla \delta u \,\mathrm{d}\Omega + \int_\Gamma \llbracket \hat{u} - u\rrbracket  \cdot \boldsymbol{\{\nabla \delta u\}} \,\mathrm{d}\Gamma + \int_{\Gamma^0} \{\hat{u} - u\} \llbracket \nabla \delta u\rrbracket  \,\mathrm{d}\Gamma^0 - \int_\Gamma \llbracket \delta u\rrbracket  \cdot \boldsymbol{\{\hat{\sigma}\}} \,\mathrm{d}\Gamma - \int_{\Gamma^0} \{\delta u\} \llbracket \hat{\sigma}\rrbracket  \,\mathrm{d}\Gamma^0 = \int_\Omega \delta u \,\mathrm{d}\Omega,\\
+#     ```
+#     The numerical fluxes chosen for the interior penalty method are $\boldsymbol{\hat{\sigma}} = \{\nabla u\} - \alpha(\llbracket u\rrbracket)$ on $\Gamma$, $\hat{u} = \{u\}$ on the interfaces between elements $\Gamma^0 : \Gamma \setminus \partial \Omega$, 
+#     and $\hat{u} = 0$ on $\partial \Omega$. Such choice results in $\boldsymbol{\{\hat{\sigma}\}} = \{\nabla u\} - \alpha(\llbracket u\rrbracket)$, $\llbracket \hat{u}\rrbracket  = 0$, $\{\hat{u}\} = \{u\}$, $\llbracket \boldsymbol{\hat{\sigma}}\rrbracket  = 0$ and the equation becomes
+#     ```math
+#      \int_\Omega \nabla u \cdot \nabla \delta u \,\mathrm{d}\Omega - \int_\Gamma \llbracket u\rrbracket  \cdot \boldsymbol{\{\nabla \delta u\}}  \,\mathrm{d}\Gamma - \int_\Gamma \llbracket \delta u\rrbracket  \cdot \boldsymbol{\{\nabla u\}} - \llbracket \delta u\rrbracket  \cdot \alpha(\llbracket u\rrbracket)  \,\mathrm{d}\Gamma = \int_\Omega \delta u \,\mathrm{d}\Omega,\\
+#     ```
+#     Where
+#     ```math
+#      \alpha(\llbracket u\rrbracket) = \mu \llbracket u\rrbracket
+#     ```
+#     Where $\mu = \eta h^{-1}$, the weak form becomes
+#     ```math
+#      \int_\Omega \nabla u \cdot \nabla \delta u \,\mathrm{d}\Omega - \int_\Gamma \llbracket u \rrbracket \cdot \boldsymbol{\{\nabla \delta u\}} + \llbracket \delta u \rrbracket  \cdot \boldsymbol{\{\nabla u\}}  \,\mathrm{d}\Gamma + \int_\Gamma \frac{\eta}{h} \llbracket u\rrbracket  \cdot \llbracket \delta u\rrbracket   \,\mathrm{d}\Gamma = \int_\Omega \delta u \,\mathrm{d}\Omega,\\
+#     ```
+# Since $\partial \Omega$ is constrained with both Dirichlet and Neumann boundary conditions the term $\int_{\partial \Omega} \nabla u \cdot \boldsymbol{n} \delta u \,\mathrm{d} \partial \Omega$ can be expressed as an integral over $\partial \Omega_N$, where $\partial \Omega_N$ is the boundaries with only prescribed Neumann boundary condition,
+# The resulting weak form is given given as follows: Find $u \in \mathbb{U}$ such that
+# ```math
+#  \int_\Omega \nabla u \cdot \nabla \delta u \,\mathrm{d}\Omega - \int_{\Gamma^0} \llbracket u\rrbracket  \cdot \boldsymbol{\{\nabla \delta u\}} + \llbracket \delta u\rrbracket  \cdot \boldsymbol{\{\nabla u\}}  \,\mathrm{d}\Gamma^0 + \int_{\Gamma^0} \frac{\eta}{h} \llbracket u\rrbracket \cdot \llbracket \delta u\rrbracket   \,\mathrm{d}\Gamma^0 = \int_\Omega \delta u \,\mathrm{d}\Omega + \int_{\partial \Omega_N} (\nabla u \cdot \boldsymbol{n}) \delta u \,\mathrm{d} \partial \Omega_N,\\
```

Comment:
@KnutAM can you please clarify on your notation proposal on this example here?

Reply:
I would write this like 
```
\int_\Omega [\nabla u] \cdot [\nabla \delta u] \,\mathrm{d}\Omega - \int_{\Gamma^0} \llbracket u\rrbracket  \cdot \{\nabla \delta u\} + \llbracket \delta u\rrbracket  \cdot \{\nabla u\}  \,\mathrm{d}\Gamma^0 + \int_{\Gamma^0} \frac{\eta}{h} \llbracket u\rrbracket \cdot \llbracket \delta u\rrbracket   \,\mathrm{d}\Gamma^0 = \int_\Omega \delta u \,\mathrm{d}\Omega + \int_{\partial \Omega_N} ([\nabla u] \cdot \boldsymbol{n}) \delta u \,\mathrm{d} \partial \Omega_N
```
to be very explicit about where the nabla-operator is acting (also removed a few bold faces that I don't think should be there - technically I even use `\boldsymbol{\nabla}` everywhere, but I don't think that is necessary since nabla is always the gradient operator).